protocols/satisfactory.js

```diff
@@ -0,0 +1,92 @@
+import Core from './core.js'
+
+/**
+ * Satisfactory servers run by default self-signed certificates. This is used to ignore the validation for the requests.
+ */
+const https = {
+  rejectUnauthorized: false
+}
+
+export default class satisfactory extends Core {
+  constructor () {
+    super()
+
+    /**
+     * To get information about the Satisfactory game server, you need to first obtain a client authenticationToken.
+     * https://satisfactory.wiki.gg/wiki/Dedicated_servers/HTTPS_API
+     */
+
+    this.authenticationToken = null
+
+    // Don't use the tcp ping probing
+    this.usedTcp = true
+
+  }
+
+  async run (state) {
+    await this.getClientAuthenticationToken()
+
+    await this.queryInfo(state)
+    await this.cleanup(state)
+  }
```

Comment:
Please do
```
const authenticationToken = await this.getAuthToken();
await this.queryInfo(state, authenticationToken);
```
to avoid having the `cleanup` function just for a single variable.

Reply:
Have updated as requested as well as developing a function for the requests as per other noted comment.